### PR TITLE
BIP32 HDNodes

### PR DESCRIPTION
**Pull request by cgcardona**
* Create BIP32 `HDNode` from seed buffer/string and extended priv/pub keys.
* Expose pubkey, privkey, extended pubkey, extended privkey and chaincode
* Derive hardened and non-hardened children
* Sign and verify messages
* Wipe private data from HDNode instance
* Full test coverage
* Full example scripts

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -56,7 +57,7 @@
     "terser-webpack-plugin": "5.1.2",
     "ts-jest": "^26.1.4",
     "ts-loader": "^6.2.2",
-    "typedoc": "0.20.36",
+    "typedoc": "0.19.2",
```
Comment (EvanJRichard) [inline package.json:60]: Any particular reason for this typedoc downgrade?

File: src/utils/hdnode.ts [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+      hdNode = new HDNode(hdKey.privateExtendedKey)
+    } else {
+      hdNode = new HDNode(hdKey.publicExtendedKey)
+    }
+    return hdNode
+  }
+
+  /**
+  * Signs the buffer hash with the private key using secp256k1 and returns the signature as a buffer.
+  * @param hash
```
Comment (EvanJRichard) [inline src/utils/hdnode.ts:38]: Should `@return` be added to this comment and above function `derive`'s comment, or do you think it's unnecessary? I could go either way.

File: src/utils/hdnode.ts [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+  sign(hash: Buffer): Buffer {
+    const sig: Buffer = this.hdkey.sign(hash)
+    return Buffer.from(sig)
+  }
+
+  /**
+  * Verifies that the signature is valid for hash and the HDNode's public key using secp256k1.
+  * @param hash 
+  * @param signature 
+  * @returns true for valid, false for invalid. Throws if the hash or signature is the wrong length.
```
Comment (EvanJRichard) [inline src/utils/hdnode.ts:49]: Similarly, would an `@throws` be appropriate?

**Review by EvanJRichard — APPROVED** [review]
Looks great! :ship: A couple non-blocking questions

File: package.json (hunk 1 above)
Comment (cgcardona) [reply package.json:60]: For some reason the `0.20.x` version doesn't work to build the AvalancheJS deps so I downgraded

```zsh
Info: Loaded plugin typedoc-plugin-external-module-name
Error: Unknown option: --mode
Error: Unknown option: --mode
npm ERR! code ELIFECYCLE
npm ERR! errno 1
npm ERR! avalanche@3.6.1 docshtml: `npx typedoc --plugin typedoc-plugin-external-module-name --mode file src`
npm ERR! Exit status 1
```

File: src/utils/hdnode.ts (hunk 2 above)
Comment (cgcardona) [reply src/utils/hdnode.ts:39]: Agreed. Resolved.

File: src/utils/hdnode.ts (hunk 3 above)
Comment (cgcardona) [reply src/utils/hdnode.ts]: Agreed. Resolved.

**Pull request merged**